announcer.rb

```diff
@@ -138,48 +149,96 @@ def play(file_data)
     elsif file_data.to_a[1]
       _, route_id = file_data.to_a[1]
       file_path = "voice/#{dir}s/#{route_id}/#{name.tr '/', '-'}.wav"
       if File.file? file_path
         system AUDIO_COMMAND, file_path
-      else say(name, route_id)
+        record_log_entry(PRESENT_TEXT_FILE, name, "route #{route_id}")
+      else say(name, "route #{route_id}")
       end
-    else say(name, route_id)
+    else say(name, dir)
+    end
+  end
+
+  def record_log_entry(log_file, message, context)
+    messages = log_entries(log_file)
+    log_entry = message
+    log_entry += " (#{context})" if context
+    return if messages.include? log_entry
+    messages << log_entry
+    File.open log_file, 'w' do |file|
+      file.puts messages.sort
     end
   end
 
+  def remove_temp_files!
+    FileUtils.rm MISSING_TEXT_FILE if File.file? MISSING_TEXT_FILE
+    FileUtils.rm PRESENT_TEXT_FILE if File.file? PRESENT_TEXT_FILE
+  end
+
   def run
     set_query_stops
     set_interval
     departures = new_departures
     announcements = departures_crossed_interval(departures, cached_departures)
     cache_departures(departures)
-    unless announcements.empty?
+    unless announcements.empty? || announcements_in_progress?
       play fragment: 'ding'
       announcements.each(&method(:make_announcement))
+      update_github_issues!
+      remove_temp_files!
     end
   end
 
-  def announce_all
-    set_query_stops
-    soonest_departures(new_departures).each do |announcement|
-      make_announcement announcement.merge(options: { exclude_stop_name: true })
+  def say(text, context)
+    system SPEECH_COMMAND, text
+    record_log_entry(PRESENT_TEXT_FILE, text, context)
+  end
+
+  def set_interval
+    return unless File.file? CONFIG_FILE
+    config = JSON.parse File.read(CONFIG_FILE)
+    @interval = config.fetch('interval') if config.key? 'interval'
+  end
+
+  def set_query_stops
+    return unless File.file? QUERY_STOPS_FILE
+    @query_stops = File.read(QUERY_STOPS_FILE).lines.map(&:strip)
+  end
+
+  def soonest_departures(departures)
+    departure_attrs = []
+    departures.each_pair do |stop_id, route_directions|
+      route_directions.each_pair do |(route_id, sign), trips|
+        departure_attrs << { route_id: route_id, headsign: sign,
+                             stop_id: stop_id, interval: trips.values.min }
+      end
     end
+    departure_attrs
   end
 
-  def say(text, context)
-    system SPEECH_COMMAND, text
-    missing_messages = if File.file? MISSING_TEXT_FILE
-                         File.read(MISSING_TEXT_FILE).lines.map(&:strip)
-                       else []
-                       end
-    log_entry = text
-    log_entry << " (#{context})" if context
-    return if missing_messages.include? log_entry
-    missing_messages << log_entry
-    File.open MISSING_TEXT_FILE, 'w' do |file|
-      file.puts missing_messages.sort
+  def update_github_issues!
+    return unless File.file? 'github_token.txt'
```

Comment:
Seems like an unneeded call to the filesystem. Could this be `return unless GITHUB_TOKEN`, or maybe `return unless defined? GITHUB_TOKEN`